server/sandbox_run_linux.go

```diff
@@ -780,14 +752,66 @@ func (s *Server) runPodSandbox(ctx context.Context, req *pb.RunPodSandboxRequest
 			}
 			g.AddMount(proc)
 		}
+		rootPair := s.defaultIDMappings.RootPair()
+		for _, path := range pathsToChown {
+			if err := os.Chown(path, rootPair.UID, rootPair.GID); err != nil {
+				return nil, errors.Wrapf(err, "cannot chown %s to %d:%d", path, rootPair.UID, rootPair.GID)
+			}
+		}
 	}
 	g.SetRootPath(mountPoint)
 
 	if os.Getenv(rootlessEnvName) != "" {
 		makeOCIConfigurationRootless(&g)
 	}
 
-	container.SetSpec(g.Config)
+	sb.SetNamespaceOptions(securityContext.GetNamespaceOptions())
+
+	spp := securityContext.GetSeccompProfilePath()
+	g.AddAnnotation(annotations.SeccompProfilePath, spp)
+	sb.SetSeccompProfilePath(spp)
+	if !privileged {
+		if err := s.setupSeccomp(ctx, &g, spp); err != nil {
+			return nil, err
+		}
+	}
+
+	runtimeType, err := s.Runtime().RuntimeType(runtimeHandler)
+	if err != nil {
+		return nil, err
+	}
+
+	// A container is kernel separated if we're using shimv2, or we're using a kata v1 binary
+	podIsKernelSeparated := runtimeType == libconfig.RuntimeTypeVM ||
```

Comment:
cc @amshinde does this condition look good? we can't strictly rely on runtime vm because we need the infra container for kata v1